snapcraft/application.py

```diff
@@ -147,11 +148,15 @@ def _configure_services(self, provider_name: str | None) -> None:
 
         super()._configure_services(provider_name)
 
-    # TODO: remove this override (#4911)
     @property
     def command_groups(self):
-        """Short-circuit the standard command groups for now."""
-        return self._command_groups
+        """Replace craft-application's LifecycleCommand group."""
+        _command_groups = super().command_groups
+        for index, command_group in enumerate(_command_groups):
+            if command_group.name == "Lifecycle":
+                _command_groups[index] = cli.CORE24_LIFECYCLE_COMMAND_GROUP
```

Comment:
In order to drop this in a future PR, I think we would have to make some upstream changes to allow overriding this: https://github.com/canonical/craft-application/blob/1509cec9af705c762b06725bf867ef1d9e1d822b/craft_application/application.py#L173.